client/components/themes-list/docs/example.jsx

```diff
@@ -0,0 +1,77 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import ThemesList from 'components/themes-list';
+
+const onScreenshotClickFn = function( theme, index ) {
+	console.log( `Theme ${ theme.id } at ${ index } clicked` );
+};
+
+const ThemesListExample = React.createClass( {
+	displayName: 'ThemesList',
+
+	render: function() {
+		const demoThemes = [ {
+			id: 'twentyfourteen',
+			name: 'Twenty Fourteen',
+			screenshot: '//i1.wp.com/theme.wordpress.com/wp-content/themes/pub/twentyfourteen/screenshot.png',
+			actionLabel: 'Click Action Theme 1',
+			onScreenshotClick: onScreenshotClickFn,
+		},
+		{	id: 'twentyfifteen',
+			name: 'Twenty Fifteen',
+			screenshot: '//i1.wp.com/theme.wordpress.com/wp-content/themes/pub/twentyfifteen/screenshot.png',
+			actionLabel: 'Click Action Theme 2',
+			onScreenshotClick: onScreenshotClickFn,
+		},
+		{	id: 'twentysixteen',
+			name: 'Twenty Sixteen',
+			screenshot: '//i1.wp.com/theme.wordpress.com/wp-content/themes/pub/twentysixteen/screenshot.png',
+			actionLabel: 'Click Action Theme 3',
+			onScreenshotClick: onScreenshotClickFn,
+		} ];
+
+		return (
+			<div className="design-assets__group">
+				<h2>
+					<a href="/devdocs/app-components/themes-list">Themes List</a>
+				</h2>
+				<ThemesList
+				themes={ demoThemes }
+				fetchNextPage={ function() {
+				} }
+				getButtonOptions={ function() {
+					return {
+						action1: {
+							label: 'Menu Item 1',
+							action: function() {
+								console.log( 'Menu Item 1 selected' );
+							}
+						},
+						action2: {
+							label: 'Menu Item 2',
+							action: function() {
+								console.log( 'Menu Item 2 selected' );
+							}
+						}
+					}
+				} }
+				getActionLabel={ function( theme ) {
+					return theme.actionLabel;
+				} }
+				onScreenshotClick={ function( theme, index ) {
+					return theme.onScreenshotClick( theme, index );
+				} }
+				/>
+			</div>
+		);
+	}
+
+} );
```

Comment:
Also, all the functions provided to `ThemesList` are unnecessarily re-instantiated on each render, so let's make them method of `ThemesListExample` instead and pass them to `ThemesList`.